src/executors/ForEachExecutor.js

```diff
@@ -0,0 +1,102 @@
+import ExecutorAbstract from '../ExecutorAbstract';
+import ExecutorFactory from '../ExecutorFactory';
+import {wrapIfNotInstance} from '../util/type';
+import {getInjections} from '../util/executors';
+
+class ForEachExecutor extends ExecutorAbstract {
+  //-----------------------------------
+  // Constructor
+  //-----------------------------------
+  /**
+   * Constructor for a new ForEachExecutor.
+   *
+   * @param thread The parent thread that owns this executor.
+   * @param arrayProperty The property (found via the available injections) to do a forEach on.
+   * @param property The name property for each element in the loop to add to the event detail for the executor.
+   * @param executor  The executor to run for each item.
+   * @param sequential True if we should wait for each previous executor to run before running the next. Default is true.
```

Comment:
JSdoc for default values is [sequential=true].
http://usejsdoc.org/tags-param.html#optional-parameters-and-default-values